tensorflow/tools/ci_build/builds/pip.sh

```diff
@@ -69,17 +69,22 @@ abs_path() {
     [[ $1 = /* ]] && echo "$1" || echo "$PWD/${1#./}"
 }
 
+# Exit after a failure
+die() {
+    echo $@
+    exit 1
+}
+
 # Get the command line arguments
 CONTAINER_TYPE=$( echo "$1" | tr '[:upper:]' '[:lower:]' )
 
 PIP_BUILD_TARGET="//tensorflow/tools/pip_package:build_pip_package"
 if [[ ${CONTAINER_TYPE} == "cpu" ]]; then
-  bazel build -c opt ${PIP_BUILD_TARGET}
+  bazel build -c opt ${PIP_BUILD_TARGET} || die "Build failed."
 elif [[ ${CONTAINER_TYPE} == "gpu" ]]; then
-  bazel build -c opt --config=cuda ${PIP_BUILD_TARGET}
+  bazel build -c opt --config=cuda ${PIP_BUILD_TARGET} || die "Build failed."
 else
-  echo "Unrecognized container type: \"${CONTAINER_TYPE}\""
-  exit 1
+  die "Unrecognized container type: \"${CONTAINER_TYPE}\""
 fi
```

Comment:
I really prefer the default to be the other way (set -e), such that if something fail, I get to know, instead of saying everywhere that if stuff fails, please tell me. You are correct, if it gets more complex, catching all of the if/fi construct may be a good idea.